llvm/lib/CodeGen/LiveVariables.cpp

```diff
@@ -277,11 +277,22 @@ void LiveVariables::HandlePhysRegUse(Register Reg, MachineInstr &MI) {
           continue;
         if (PartDefRegs.count(SubReg))
           continue;
+
+        // Check if SubReg is defined at LastPartialDef.
+        bool IsDefinedHere = false;
+        for (int I = 0; I < LastPartialDef->getNumOperands(); ++I) {
```

Comment:
This whole function looks like it could use a rewrite. Is this a re-invention of MachineInstr::modifiesRegister?

Reply:
The parent function, `LiveVariables::HandlePhysRegUse`, will additionally collect definitions of `Reg` (e.g. four `COPY` instructions for a use of `$sgpr0_$sgpr1_$sgpr2_sgpr3` in a subsequent instruction), and mark the last definition with its implicit uses and defs.

`modifiesRegister` can be used here in place of this loop.